sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -669,9 +669,7 @@ object GpuOverrides extends Logging {
       case FloatType => true
       case DoubleType => true
       case DateType => true
-      case TimestampType =>
-        TypeChecks.areTimestampsSupported(ZoneId.systemDefault()) &&
-        TypeChecks.areTimestampsSupported(SQLConf.get.sessionLocalTimeZone)
+      case TimestampType => true
```

Comment:
Do we need to consider the timezone check for scan and writer parts? AFAIK, when scanning data from Parquet, ```spark.sql.session.timeZone``` is supposed to be respect.

If applies, we should add some python tests as well.

Reply:
This check is used by `InternalColumnarRddConverter` and `HostToGpuCoalesceIterator`.
`Coalesce` can handle non UTC timestamp. But not sure `InternalColumnarRddConverter`, seems it's also OK.

Reply:
I think we will need to check these. For me, anything that does not have a test that shows it works fully in at least one other time zone must fall back to the CPU if it sees a timestamp that is not UTC.

Parquet for example has the rebase mode for older timestamps that requires knowing the timezone to do properly.